tmail-backend/mailbox/encrypted/cassandra/src/main/java/com/linagora/tmail/encrypted/cassandra/table/CassandraEncryptedEmailContentStore.scala

```diff
@@ -17,25 +23,43 @@ case class EncryptedEmailContentStoreCassandraModule() extends AbstractModule {
 
     Multibinder.newSetBinder(binder, classOf[CassandraModule])
       .addBinding()
-      .toInstance(CassandraEncryptedEmailStoreModule.EMAIL_CONTENT_MODULE)
-
-    Multibinder.newSetBinder(binder, classOf[CassandraModule])
-      .addBinding()
-      .toInstance(CassandraEncryptedEmailStoreModule.MESSAGE_ID_BLOB_ID_MODULE)
+      .toInstance(CassandraEncryptedEmailStoreModule.MODULE)
   }
 }
 
 class CassandraEncryptedEmailContentStore @Inject()(blobStore: BlobStore,
-                                                    encryptedEmailDAO: CassandraEncryptedEmailDAO,
-                                                    encryptedEmailIdBlobIdDAO: CassandraEncryptedEmailIdBlobIdDAO) extends EncryptedEmailContentStore {
+                                                    bucketName: BucketName,
+                                                    storagePolicy: StoragePolicy,
+                                                    encryptedEmailDAO: CassandraEncryptedEmailDAO) extends EncryptedEmailContentStore {
+
+  override def store(messageId: MessageId, encryptedEmailContent: EncryptedEmailContent): Publisher[Unit] = {
+    val position: AtomicInteger = new AtomicInteger(POSITION_NUMBER_START_AT)
+    SFlux.fromIterable(encryptedEmailContent.encryptedAttachmentContents)
+      .concatMap(encryptedAttachmentContent => SMono.fromPublisher(blobStore.save(bucketName, encryptedAttachmentContent, storagePolicy)))
+      .map(blobId => position.getAndIncrement() -> blobId)
+      .collectSeq()
+      .flatMap(positionBlobIdList => encryptedEmailDAO.insert(messageId,
+        EncryptedEmailDetailedView.from(encryptedEmailContent),
+        positionBlobIdList.toMap))
+      .`then`()
+  }
 
-  override def store(messageId: MessageId, encryptedEmailContent: EncryptedEmailContent): Unit = ???
+  override def delete(messageId: MessageId): Publisher[Unit] =
+    SFlux.zip(deleteBlobStore(messageId), encryptedEmailDAO.delete(messageId))
+      .`then`()
 
-  override def delete(messageId: MessageId): Unit = ???
+  override def retrieveFastView(messageId: MessageId): Publisher[EncryptedEmailFastView] =
+    encryptedEmailDAO.get(messageId)
+      .map(encryptedEmailDetailedView => EncryptedEmailFastView.from(encryptedEmailDetailedView))
 
-  override def retrieveFastView(messageId: MessageId): EncryptedEmailFastView = ???
+  override def retrieveDetailedView(messageId: MessageId): Publisher[EncryptedEmailDetailedView] =
+    encryptedEmailDAO.get(messageId)
 
-  override def retrieveDetailedView(messageId: MessageId): EncryptedEmailDetailedView = ???
+  override def retrieveAttachmentContent(messageId: MessageId, position: Int): Publisher[BlobId] =
+    encryptedEmailDAO.getBlobId(messageId, position)
+      .onErrorResume((error)=>throw AttachmentNotFoundException(messageId, position))
```

Comment:
I have a reactor problem here
When run test
```
Expecting actual throwable to be an instance of:
  com.linagora.tmail.encrypted.AttachmentNotFoundException
but was:
  reactor.core.Exceptions$ReactiveException: com.linagora.tmail.encrypted.AttachmentNotFoundException
```

Reply:
Hmm not really...

You can switch the assertion in the tests to hasCause for instance.

BTW you can use onErrorMap to aoid this uggy throw I imagine...

Reply:
I fixed it,  `AttachmentNotFoundException extends Exception`  => `AttachmentNotFoundException extends RuntimeException`.